src/commands/deployment/correlate.ts

```diff
@@ -0,0 +1,169 @@
+    }
+    const ciEnv: Record<string, string> = {
+      ...tags,
+      ...envVars,
+    }
+
+    const git = simpleGit({
+      baseDir: process.cwd(),
+      binary: 'git',
+      maxConcurrentProcesses: 2, // max 2 git commands at the same time
+    })
+
+    const currentBranch = await gitCurrentBranch(git)
+    if (!currentBranch) {
+      this.logger.error('Could not get current branch')
+
+      return 1
+    }
+
+    let localCommitShas: string[]
+    if (this.configurationRepo) {
+      localCommitShas = await gitLocalCommitShas(git, currentBranch)
+    } else {
+      ;[this.configurationRepo, localCommitShas] = await Promise.all([
+        gitRepositoryURL(git),
+        gitLocalCommitShas(git, currentBranch),
+      ])
+    }
+
+    if (this.configurationRepo === undefined || this.configurationRepo === '') {
+      this.logger.error('Could not retrieve repository URL, check out a repository or provide it with --config-repo')
+
+      return 1
+    }
+
+    await this.sendCorrelationData(ciEnv[CI_PROVIDER_NAME], localCommitShas, ciEnv, this.config.apiKey)
+  }
+
+  private async sendCorrelationData(
+    ciProvider: string,
+    configCommitShas: string[],
+    ciEnv: Record<string, string>,
+    apiKey: string
+  ) {
+    const correlateEvent = {
+      type: 'ci_app_deployment_correlate',
+      attributes: {
+        ci_provider: ciProvider,
+        cd_provider: this.cdProviderParam,
+        config_repo_url: this.configurationRepo,
+        config_commit_shas: configCommitShas,
+        ci_env: ciEnv,
+      },
+    }
+
+    if (this.dryRun) {
+      this.logger.info(`[DRYRUN] Sending correlation event\n data: ` + JSON.stringify(correlateEvent, undefined, 2))
+
+      return
+    }
+
+    const site = process.env.DATADOG_SITE || process.env.DD_SITE || 'datadoghq.com'
+    const baseAPIURL = `https://${getApiHostForSite(site)}`
+    const request = getRequestBuilder({baseUrl: baseAPIURL, apiKey})
+    const doRequest = () =>
+      request({
+        data: {
+          data: correlateEvent,
+        },
+        method: 'post',
+        url: '/api/v2/ci/deployments/correlate',
+      })
+
+    try {
+      await retryRequest(doRequest, {
+        maxTimeout: 30000,
+        minTimeout: 5000,
+        onRetry: (e, attempt) => {
+          this.logger.warn(
+            `[attempt ${attempt}] Could not send deployment correlation data. Retrying...: ${e.message}\n`
+          )
+        },
+        retries: 5,
+      })
+    } catch (error) {
+      this.logger.error(`Failed to send deployment correlation data: ${error.message}`)
+    }
+  }
+
+  private validateTags(tags: Record<string, string>): boolean {
+    if (!tags[GIT_REPOSITORY_URL]) {
+      this.logger.error('Could not extract the source code repository URL from the CI environment variables')
+
+      return false
+    }
+    if (!tags[GIT_SHA]) {
+      this.logger.error('Could not extract the source git commit sha from the CI environment variables')
```

Comment:
**nit** I think there is no need to be so "exact" in the error that is customer facing 😁 
```suggestion
      this.logger.error('Could not extract the commit SHA from the CI environment variables')
```